src/Features/CSharp/Portable/InlineHints/CSharpInlineTypeHintsService.cs

```diff
@@ -80,6 +80,15 @@ public CSharpInlineTypeHintsService()
                 }
             }
 
+            if (node is ImplicitObjectCreationExpressionSyntax { Parent: ArgumentSyntax } implicitNew)
```

Comment:
So why restrict to just things being passed as arguments here? Isn't this something you might potentially want for all target-typed new expressions, at least if there's the "display all" thing going on?

Reply:
(I'm also wondering if the code we have in _parameter_ hints needs an update, because that might be only looking at regular 'new' expressions which might be bugged as well.)

Reply:
> So why restrict to just things being passed as arguments here

Issue #49657 explicit excluded other cases, as most of the time the type is close to `new()`

Reply:
> (I'm also wondering if the code we have in parameter hints needs an update, because that might be only looking at regular 'new' expressions which might be bugged as well.)

AbstractInlineParameterNameHintsService displays the parameter name only, so I think there is no need to change it. There is a `InlineHintsOptions.ForObjectCreationParameters` though, which could be used to suppress the type hint here.